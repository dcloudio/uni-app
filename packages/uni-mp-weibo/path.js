const path = require('path')
module.exports = {
  src: path.join(__dirname, 'src'),
  lib: path.join(__dirname, 'lib')
}
