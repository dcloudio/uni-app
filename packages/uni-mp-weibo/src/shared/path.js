export function parsePath (path) {
  let hash = ''
  let query = ''

  const hashIndex = path.indexOf('#')
  if (hashIndex >= 0) {
    hash = path.slice(hashIndex)
    path = path.slice(0, hashIndex)
  }

  const queryIndex = path.indexOf('?')
  if (queryIndex >= 0) {
    query = path.slice(queryIndex + 1)
    path = path.slice(0, queryIndex)
  }

  return {
    path,
    query,
    hash
  }
}
