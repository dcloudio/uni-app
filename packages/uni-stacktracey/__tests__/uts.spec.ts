import path from 'path'

const {
  generateCodeFrameWithKotlinStacktrace,
  generateCodeFrameWithSwiftStacktrace,
} = require('../dist/uni-stacktracey.cjs.js')

const utsProjectDir = path.resolve(__dirname, '../test/uts')

describe('code-frame-uts', () => {
  test('android', async () => {
    expect(
      await generateCodeFrameWithKotlinStacktrace(androidUniModulesError, {
        name: 'uni_modules/test-uts1',
        inputDir: '/Users/fxy/DCloud/test-uts',
        outputDir: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
      })
    ).toMatchSnapshot()
    expect(
      await generateCodeFrameWithKotlinStacktrace(androidUtssdkError, {
        name: 'utssdk/test2',
        inputDir: '/Users/fxy/DCloud/test-uts',
        outputDir: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
      })
    ).toMatchSnapshot()
  })
  test('ios', async () => {
    expect(
      await generateCodeFrameWithSwiftStacktrace(iosUniModulesError, {
        name: 'uni_modules/test-uts1',
        inputDir: '/Users/fxy/DCloud/test-uts',
        outputDir: path.resolve(utsProjectDir, 'unpackage/dist/dev/app-plus'),
      })
    ).toMatchSnapshot()
  })
})

const androidUniModulesError = `e: uni_modules/test-uts1/utssdk/app-android/index.kt: (8, 12): The integer literal does not conform to the expected type String

FAILURE: Build failed with an exception.`

const androidUtssdkError = `e: utssdk/test2/app-android/index.kt: (8, 12): The integer literal does not conform to the expected type String

FAILURE: Build failed with an exception.`

const iosUniModulesError = `CompileSwift normal armv7 uni_modules/test-uts1/utssdk/app-ios/src/index.swift (in target 'uni_modules_test_uts1' from project 'UTS')
uni_modules/test-uts1/utssdk/app-ios/src/index.swift:3:12: error: cannot convert return expression of type 'Int' to return type 'String'

CompileSwift normal arm64 uni_modules/test-uts1/utssdk/app-ios/src/index.swift (in target 'uni_modules_test_uts1' from project 'UTS')
uni_modules/test-uts1/utssdk/app-ios/src/index.swift:3:12: error: cannot convert return expression of type 'Int' to return type 'String'`
