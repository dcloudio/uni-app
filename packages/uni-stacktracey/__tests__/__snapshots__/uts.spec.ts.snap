// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-frame-uts android 1`] = `
"error: The integer literal does not conform to the expected type String
at uni_modules/test-uts1/utssdk/index.uts:2:10
1  |  export function test(): string {
2  |    return 1;
   |            ^
3  |  }
4  |  


FAILURE: Build failed with an exception."
`;

exports[`code-frame-uts android 2`] = `
"error: The integer literal does not conform to the expected type String
at utssdk/test2/index.uts:2:10
1  |  export function test(): string {
2  |    return 2;
   |            ^
3  |  }
4  |  


FAILURE: Build failed with an exception."
`;

exports[`code-frame-uts ios 1`] = `
"CompileSwift normal armv7 uni_modules/test-uts1/utssdk/app-ios/src/index.swift (in target 'uni_modules_test_uts1' from project 'UTS')
error: cannot convert return expression of type 'Int' to return type 'String'
at uni_modules/test-uts1/utssdk/app-ios/index.uts:2:10
1  |  export function test(): string {
2  |    return 1;
   |            ^
3  |  }
4  |  


CompileSwift normal arm64 uni_modules/test-uts1/utssdk/app-ios/src/index.swift (in target 'uni_modules_test_uts1' from project 'UTS')
error: cannot convert return expression of type 'Int' to return type 'String'
at uni_modules/test-uts1/utssdk/app-ios/index.uts:2:10
1  |  export function test(): string {
2  |    return 1;
   |            ^
3  |  }
4  |  
"
`;
